appium-dotnet-driver/Appium/Tizen/TizenDriver.cs

```diff
@@ -64,24 +86,88 @@ public TizenDriver(Uri remoteAddress, DesiredCapabilities desiredCapabilities)
+        /// </summary>
+        /// <param name="remoteAddress">URI containing the address of the WebDriver remote server (e.g. http://127.0.0.1:4723/wd/hub).</param>
+        /// <param name="desiredCapabilities">An <see cref="DesiredCapabilities"/> object containing the desired capabilities.</param>
+        /// <param name="commandTimeout">The maximum amount of time to wait for each command.</param>
         public TizenDriver(Uri remoteAddress, DesiredCapabilities desiredCapabilities, TimeSpan commandTimeout)
             : base(remoteAddress, SetPlatformToCapabilities(desiredCapabilities, Platform), commandTimeout)
         {
         }
 
-        // TODO: This feature may not work properly at the moment.
+        /// <summary>
+        /// Initializes a new instance of the TizenDriver class using the specified Appium local service, desired capabilities, and command timeout.
+        /// </summary>
+        /// <param name="service">the specified Appium local service</param>
+        /// <param name="desiredCapabilities">An <see cref="ICapabilities"/> object containing the desired capabilities.</param>
+        /// <param name="commandTimeout">The maximum amount of time to wait for each command.</param>
         public TizenDriver(AppiumLocalService service, DesiredCapabilities desiredCapabilities, TimeSpan commandTimeout)
             : base(service, SetPlatformToCapabilities(desiredCapabilities, Platform), commandTimeout)
         {
         }
 
+        public bool CompareImages(Image firstImage, Image secondImage)
+        {
+            if (firstImage == null || secondImage == null)
+            {
+                throw new NullReferenceException("Screenshot and Rectangle should not be null");
+            }
+
+            var firstBitmap = new Bitmap(firstImage);
+            var secondBitmap = new Bitmap(secondImage);
+            Color firstPixel;
+            Color secondPixel;
+            int firstGrayScale = 0;
+            int secondGrayScale = 0;
+
+            for (int i = 0; i < firstImage.Width; i++)
+            {
+                for (int j = 0; j < firstImage.Height; j++)
+                {
+                    firstPixel = firstBitmap.GetPixel(i, j);
+                    secondPixel = secondBitmap.GetPixel(i, j);
+
+                    firstGrayScale = (int)((firstPixel.R * 0.3) + (firstPixel.G * 0.59) + (firstPixel.B * 0.11));
+                    secondGrayScale = (int)((secondPixel.R * 0.3) + (secondPixel.G * 0.59) + (secondPixel.B * 0.11));
+
+                    int diff = firstGrayScale - secondGrayScale;
+
+                    if (Math.Abs(diff) > 3)
+                    {
+                        return false;
+                    }
+                }
+            }
+
+            return true;
+        }
+
+        public Image TransformScreenshot(Screenshot screenshot, Rectangle rectangle)
+        {
+            if (screenshot == null || rectangle == null)
```

Comment:
I think rectangle is safe. Bitmap.Clone() will do instead.

Reply:
Done